pkg/k8s/admission-webhook/validating-webhook.go

```diff
@@ -0,0 +1,336 @@
+		deserializer             = codecs.UniversalDeserializer()
+		requestedAdmissionReview admissionv1.AdmissionReview
+	)
+	w.requestBody = requestBody
+	admissionv1.AddToScheme(scheme)
+
+	// decode incoming admission request
+	_, _, err := deserializer.Decode(requestBody, nil, &requestedAdmissionReview)
+	if err != nil {
+		errMsg := "failed to decode validating admission webhook request body"
+		zap.S().Error(errMsg, zap.Error(err))
+		return requestedAdmissionReview, fmt.Errorf("%s, error: %w", errMsg, err)
+	}
+
+	return requestedAdmissionReview, nil
+}
+
+// ProcessWebhook processes the incoming AdmissionReview and creates
+// a response
+func (w ValidatingWebhook) ProcessWebhook(review admissionv1.AdmissionReview, serverURL string) (*admissionv1.AdmissionReview, error) {
+
+	var (
+		output         runtime.Output
+		denyViolations []results.Violation
+		logURL         = w.dblogger.GetLogURL(serverURL, string(review.Request.UID))
+		allowed        = false
+		errMsg         = "%s; error: %w"
+	)
+
+	// In case the object is nil => an operation of DELETE happened, just return 'allow' since there is nothing to check
+	if len(review.Request.Object.Raw) < 1 {
+		zap.S().Info(ErrEmptyAdmissionReview, zap.Any("admission review object", review))
+		return w.createResponseAdmissionReview(review, true, output, logURL), ErrEmptyAdmissionReview
+	}
+
+	// Save the object into a temp file for the policy engines
+	tempFile, err := utils.CreateTempFile(review.Request.Object.Raw, "json")
+	defer os.Remove(tempFile.Name())
+	if err != nil {
+		msg := "failed to create temp file for validating admission review request"
+		zap.S().Error(msg, zap.Error(err))
+		return w.createResponseAdmissionReview(review, allowed, output, logURL), fmt.Errorf(errMsg, msg, err)
+	}
+
+	// Run the policy engines
+	output, err = w.scanK8sFile(tempFile.Name())
+	if err != nil {
+		msg := "failed to evaluate terrascan policies"
+		zap.S().Errorf(msg, zap.Error(err))
+		return w.createResponseAdmissionReview(review, allowed, output, logURL), fmt.Errorf(errMsg, msg, err)
+	}
+
+	// Calculate if there are anydeny violations
+	denyViolations, err = w.getDenyViolations(output)
+	if err != nil {
+		msg := "failed to figure out denied violations"
+		zap.S().Errorf(msg, zap.Error(err))
+		return w.createResponseAdmissionReview(review, allowed, output, logURL), fmt.Errorf(errMsg, msg, err)
+	}
+	allowed = len(denyViolations) < 1
+
+	// Log the request in the DB
+	err = w.logWebhook(output, string(review.Request.UID), denyViolations, allowed)
+	if err != nil {
+		msg := "failed to log validating admission review request into database"
+		zap.S().Error(msg, zap.Error(err))
+	}
+
+	return w.createResponseAdmissionReview(review, allowed, output, logURL), nil
+}
+
+func (w ValidatingWebhook) scanK8sFile(filePath string) (runtime.Output, error) {
+
+	var (
+		executor *runtime.Executor
+		err      error
+		result   runtime.Output
+	)
+
+	if flag.Lookup("test.v") != nil {
```

Comment:
Not looking for a change, but this feels strange as almost all the code uses Cobra...